### PR TITLE
Update rule S5122 [java/js] taint/user-controlled code examples

### PR DESCRIPTION
**Pull request by eric-therond-sonarsource**

### CODE REVIEW

File: rules/S5122/java/rule.adoc [hunk 1]
```diff
@@ -99,6 +111,21 @@ class Safe implements WebMvcConfigurer {
...
 ----
 
+User-controlled origin validated with an allow-list:
+
+----
+public ResponseEntity<String> userControlledOrigin(@RequestHeader("Origin") String origin) 
+{
+  HttpHeaders responseHeaders = new HttpHeaders();
+  if (trustedOrigins.contains(origin))
+  {
```
Comment (hendrik-buchwald-sonarsource) [inline rules/S5122/java/rule.adoc:121]: Minor: in the other Java code samples it seems to be rather usual to not have the opening curly brace in a new line.

**Review by hendrik-buchwald-sonarsource — APPROVED** [review]
Interesting choice to combine Java and JS but I guess that is fine?

File: rules/S5122/java/rule.adoc (hunk 1 above)
Comment (eric-therond-sonarsource) [reply rules/S5122/java/rule.adoc:121]: +1 I will adjust to comply with the rest of the rspec code samples

**Comment by eric-therond-sonarsource** [comment]
Yes I think it doesn't matter, this rule will likely be implemented for all languages at the same time for SonarSecurity
and if it is not the case, we can manage that with [the related process](https://github.com/SonarSource/rspec#multi-language-rule-creation)

File: rules/S5122/java/rule.adoc [hunk 2]
```diff
@@ -52,6 +49,17 @@ class Insecure implements WebMvcConfigurer {
 }
 ----
 
+User-controlled origin:
+
```
Comment (loris-s-sonarsource) [inline rules/S5122/java/rule.adoc:53]:
```suggestion

[source,java]
```

File: rules/S5122/javascript/rule.adoc [hunk 3]
```diff
@@ -33,6 +33,15 @@ let app2 = express();
 app2.use(cors(corsOptions));
 ----
 
+User-controlled origin:
+
```
Comment (loris-s-sonarsource) [inline rules/S5122/javascript/rule.adoc:37]:
```suggestion

[source,javascript]
```

File: rules/S5122/javascript/rule.adoc [hunk 4]
```diff
@@ -59,6 +68,18 @@ let app = express();
 app.use(cors(corsOptions));
 ----
 
+User-controlled origin validated with an allow-list:
+
```
Comment (loris-s-sonarsource) [inline rules/S5122/javascript/rule.adoc:72]:
```suggestion

[source,javascript]
```

**Pull request merged**